src/WinRT.Runtime/Projections/Bindable.net5.cs

```diff
@@ -396,22 +398,84 @@ internal unsafe interface IEnumerable : global::System.Collections.IEnumerable,
 #pragma warning disable CA2257 // This member is a type (so it cannot be invoked)
         public sealed class AdaptiveFromAbiHelper : FromAbiHelper, global::System.Collections.IEnumerable
 #pragma warning restore CA2257
-        {
-            private readonly Func<IWinRTObject, global::System.Collections.IEnumerator> _enumerator;
-
+        {
+#if NET8_0_OR_GREATER
+            private readonly MethodInvoker _enumerator;
+#else
+            private readonly MethodInfo _enumerator;
+#endif
+
+#if NET
+            [DynamicDependency(DynamicallyAccessedMemberTypes.PublicMethods, typeof(IEnumerable<>))]
+            [SuppressMessage("Trimming", "IL2070", Justification = "We're using '[DynamicDependency]' to preserve public methods of 'IEnumerable<T>'.")]
+            [SuppressMessage("Trimming", "IL2075", Justification = "We're using '[DynamicDependency]' to preserve public methods of 'IEnumerable<T>'.")]

```

Comment:
This suppression/dynamicdependency combo is avoidable. Replace the `var methodInfo` line with

```csharp
var methodInfo = enumGenericType.GetMemberWithSameMetadataDefinitionAs(typeof(IEnumerable<>).GetMethod("GetEnumerator"));
```

Reply:
TIL, that's neat! 👀

Out of curiosity, what happens if the type implements two `IEnumerable<T>`-s? Will this just throw?

Reply:
I don't understand the question. You call the API on `IEnumerable<SomeType>` with `IEnumerable<>.Method` and it returns `IEnumerable<SomeType>.Method`.

Reply:
Oh my bad, yeah that made no sense, I misread your comment. I can see that the call is actually on the constructed interface type, not on the concrete type that implements the interface. Ignore that 😅